frontend/lib/features/pfis/pfis_page.dart

```diff
@@ -3,32 +3,40 @@ import 'package:flutter_starter/features/pfis/pfi_providers.dart';
 import 'package:flutter_starter/features/pfis/pfi_verification_page.dart';
 import 'package:flutter_starter/l10n/app_localizations.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:flutter_starter/features/pfis/pfi.dart';
 
 class PfisPage extends HookConsumerWidget {
   const PfisPage({super.key});
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final pfis = ref.watch(pfisProvider);
+    // Watch the provider and get the AsyncValue object
+    AsyncValue<List<Pfi>> pfisAsyncValue = ref.watch(pfisProvider);
+
     return Scaffold(
       appBar: AppBar(title: Text(Loc.of(context).selectYourRegion)),
-      body: ListView(
-        children: [
-          ...pfis.map(
-            (pfi) => ListTile(
-              title: Text(pfi.name),
-              subtitle: Text(pfi.didUri),
-              trailing: const Icon(Icons.chevron_right),
-              onTap: () async {
-                Navigator.of(context).push(
-                  MaterialPageRoute(
-                    builder: (_) => PfiVerificationPage(pfi: pfi),
-                  ),
-                );
-              },
-            ),
-          )
-        ],
+      body: pfisAsyncValue.when(
```

Comment:
nit: but you can use the `ref` directly here. This has the advantage that only this part of the widget tree will rebuild when the value changes.

```dart
ref.watch(pfisProvider).when(....
```

Reply:
yep - did it